### PR TITLE
Adding an admin page to manage a specific user's status

### PR DESCRIPTION
**Pull request by fozziethebeat**
This includes:
- A new web API route for updating the user's role
- A new page that displays a user's data in a form and allows updating the role via a selector
- Use of a new form library that can support error handling and supposedly simplifies things (Formik)
- A prisma seed library to inject a bunch of fake users into the database

Not done yet: 
- pagination in the admin user page
- nice styling.  The manage user page looks...odd

### CODE REVIEW

File: website/src/pages/api/admin/update_user.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * Update's the user's data in the database.  Accessible only to admins.
+ */
+const handler = async (req, res) => {
+  const token = await getToken({ req });
+
+  // Return nothing if the user isn't registered or if the user isn't an admin.
+  if (!token || token.role !== "admin") {
+    res.status(403).end();
+    return;
+  }
```
Comment (AbdBarho) [inline website/src/pages/api/admin/update_user.ts:14]: could we extract these to a middleware?

maybe something like:


```ts
const handler = withAdmin(async (token, req, res) => {
...
});
```
or maybe
```ts
```ts
const handler = withRole("admin", async (token, req, res) => {
...
});
```

**Review by AbdBarho — CHANGES_REQUESTED** [review]
Formik is okay, I used it one or two times before.

File: website/src/pages/admin/manage_user/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+          name="description"
+          content="Conversational AI for everyone. An open source project to create a chat enabled GPT LLM run by LAION and contributors around the world."
+        />
+      </Head>
+      <Container className="oa-basic-theme">
+        <Formik
+          initialValues={user}
+          onSubmit={(values) => {
+            trigger(values);
+          }}
```
[the remark this answers is not in the document]
Comment (fozziethebeat) [reply website/src/pages/admin/manage_user/[id].tsx:66]: `npm run build` *really* didn't like that due to some type issues.  Will leave as is for now.

File: website/src/pages/admin/manage_user/[id].tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        duration: 1000,
+        isClosable: true,
+      });
+    },
+  });
+
+  return (
+    <>
+      <Head>
+        <title>Open Assistant</title>
```
[the remark this answers is not in the document]
Comment (fozziethebeat) [reply website/src/pages/admin/manage_user/[id].tsx]: Done.

File: website/src/pages/api/admin/update_user.ts (hunk 1 above)
Comment (fozziethebeat) [reply website/src/pages/api/admin/update_user.ts]: Done.  I looked at using the actual `middleware.ts` but it looks a bit too messy.  A wrapper is a lot cleaner.  makes me miss RedwoodJS where they built this into the graphQL handler.

**Review by AbdBarho — APPROVED** [review]

**Pull request merged**